satpy/readers/fci_l1c_nc.py

```diff
@@ -700,6 +709,31 @@ def calibrate_rad_to_refl(self, radiance, key):
         res = 100 * radiance * np.float32(np.pi) * np.float32(sun_earth_distance) ** np.float32(2) / cesi
         return res
 
+    def _collect_listed_variables(self, file_handle, listed_variables, filetype_info):
+        listed_variables = self._recycle_shared_info(listed_variables, filetype_info)
+        super()._collect_listed_variables(file_handle, listed_variables, filetype_info)
+        self._store_shared_info(filetype_info)
+
+    def _recycle_shared_info(self, listed_variables, filetype_info):
```

Comment:
When I see the word `recycle`, I think of garbage collection, but I don't think that's what you mean here.  Maybe something like `collect_from_inter_segment_cache`, if I understand correctly what your intention is?

Reply:
I had the same thought earlier today, that I'm not recycling but reusing the data. I'll think of a better name for this.